### PR TITLE
BaSP-TG-32: add and edit Super Admin

### PR DESCRIPTION
**Pull request by camilavaleriarojas**
Add, modal and edit functionalities implemented to Super Admin entity.

### CODE REVIEW

File: src/Components/SuperAdmins/List/index.js [hunk 1]
```diff
@@ -21,9 +25,27 @@ const SuperAdminsList = (props) => {
...
+                <td
+                  onClick={() => {
+                    onClickSuperAdmin(superAdmin._id);
+                  }}
+                >
+                  {superAdmin.email}
+                </td>
                 <td>
                   <button
                     className={styles.button}
```
Comment (lmalvasia) [inline src/Components/SuperAdmins/List/index.js:51]: To the onClick function, we have to add this:
```
                    onClick={(event) => {
                      event.stopPropagation();
                      props.setDeleteSuperAdmin(superAdmin._id);
                      props.setModal(true);
                    }}
```

Comment (lmalvasia) [reply src/Components/SuperAdmins/List/index.js:51]: This is going to stop the redirection to the form page :)

File: src/Components/SuperAdmins/Form/index.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
+import styles from './form.module.css';
+import React, { useState, useEffect } from 'react';
+import FormModal from './Modal/index';
+import Title from '../Form/ModalTitle/index';
+
+const Form = () => {
+  const [modalTitle, setTitle] = useState('Add new Super Admin');
+  const [autoForm, setAutoForm] = useState(false);
```
Comment (lmalvasia) [inline src/Components/SuperAdmins/Form/index.js:8]:
```suggestion
  const [isEditing, setIsEditing] = useState(false);
```

File: src/Components/SuperAdmins/Form/index.js [hunk 3]
```diff
@@ -0,0 +1,158 @@
+import styles from './form.module.css';
+import React, { useState, useEffect } from 'react';
+import FormModal from './Modal/index';
+import Title from '../Form/ModalTitle/index';
+
+const Form = () => {
+  const [modalTitle, setTitle] = useState('Add new Super Admin');
```
Comment (lmalvasia) [inline src/Components/SuperAdmins/Form/index.js:7]: This is not necessary, we could change the title based on the isEditing prop (check next comments)

File: src/Components/SuperAdmins/Form/index.js [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+  const [autoForm, setAutoForm] = useState(false);
+  const url = window.location.href;
```
Comment (lmalvasia) [inline src/Components/SuperAdmins/Form/index.js:9]: Remove this. It's not necessary here.

File: src/Components/SuperAdmins/Form/index.js [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+  const [showFormModal, setFormModal] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [addSuperAdmin, setAddSuperAdmin] = useState({
```
Comment (lmalvasia) [inline src/Components/SuperAdmins/Form/index.js:12]:
```suggestion
  const [superAdmin, setSuperAdmin] = useState({
```

File: src/Components/SuperAdmins/Form/index.js [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+  const formCloseModal = () => {
+    setFormModal(false);
+  };
+
+  const onChange = (e) => {
+    setAddSuperAdmin({ ...addSuperAdmin, [e.target.name]: e.target.value });
+  };
+
+  useEffect(async () => {
+    if (url.match('id')) {
```
Comment (lmalvasia) [inline src/Components/SuperAdmins/Form/index.js:28]:
```suggestion
    const params = new URLSearchParams(window.location.search);
    const id = params.get('id');
    if (id) {
```

File: src/Components/SuperAdmins/Form/index.js [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+      const id = url.substring(url.indexOf('id') + 3);
```
Comment (lmalvasia) [inline src/Components/SuperAdmins/Form/index.js:29]: Remove this.

File: src/Components/SuperAdmins/Form/index.js [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+      try {
+        const response = await fetch(`${process.env.REACT_APP_API_URL}/super-admins/${id}`, {
+          method: 'GET'
+        });
+        const data = await response.json();
+        setTitle('Edit Super Admin');
```
Comment (lmalvasia) [inline src/Components/SuperAdmins/Form/index.js:35]: Remove this.

File: src/Components/SuperAdmins/Form/index.js [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+        setAutoForm(true);
```
Comment (lmalvasia) [inline src/Components/SuperAdmins/Form/index.js:36]:
```suggestion
        setIsEditing(true);
```

File: src/Components/SuperAdmins/Form/index.js [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+        setAddSuperAdmin({
```
Comment (lmalvasia) [inline src/Components/SuperAdmins/Form/index.js:37]:
```suggestion
        setSuperAdmin({
```

File: src/Components/SuperAdmins/Form/index.js [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+          name: data.data.name,
+          lastName: data.data.lastName,
+          email: data.data.email,
+          password: data.data.password
+        });
+      } catch (error) {
+        console.log(error);
```
Comment (lmalvasia) [inline src/Components/SuperAdmins/Form/index.js:44]:
```suggestion
        alert(error);
```

File: src/Components/SuperAdmins/Form/index.js [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+      }
+    }
+  }, []);
+
+  const onSubmit = async (e) => {
+    if (!autoForm) {
```
Comment (lmalvasia) [inline src/Components/SuperAdmins/Form/index.js:50]:
```suggestion
    if (!isEditing) {
```

File: src/Components/SuperAdmins/Form/index.js [hunk 13]
```diff
@@ -0,0 +1,158 @@
...
+        })
+      });
+      const content = await res.json();
+      if (!content.error) {
+        window.location.assign('/super-admins');
+      } else {
+        setFormModal(true);
+        setErrorMessage(content.message);
+      }
+      console.log(content);
```
Comment (lmalvasia) [inline src/Components/SuperAdmins/Form/index.js:72]: Remove this console log

File: src/Components/SuperAdmins/Form/index.js [hunk 14]
```diff
@@ -0,0 +1,158 @@
...
+        });
+      } catch (error) {
+        console.log(error);
+      }
+    }
+  }, []);
+
+  const onSubmit = async (e) => {
+    if (!autoForm) {
+      e.preventDefault();
```
Comment (lmalvasia) [inline src/Components/SuperAdmins/Form/index.js:51]: Move this prevent default to the top of the function.

File: src/Components/SuperAdmins/Form/index.js [hunk 15]
```diff
@@ -0,0 +1,158 @@
...
+      const content = await res.json();
+      if (!content.error) {
+        window.location.assign('/super-admins');
+      } else {
+        setFormModal(true);
+        setErrorMessage(content.message);
+      }
+      console.log(content);
+    } else {
+      e.preventDefault();
```
Comment (lmalvasia) [inline src/Components/SuperAdmins/Form/index.js:74]: Remove this one, we could have only one at the top of the function.

File: src/Components/SuperAdmins/Form/index.js [hunk 16]
```diff
@@ -0,0 +1,158 @@
...
+      const id = url.substring(url.indexOf('id') + 3);
```
Comment (lmalvasia) [inline src/Components/SuperAdmins/Form/index.js:75]: const params = new URLSearchParams(window.location.search);
const id = params.get('id');

File: src/Components/SuperAdmins/Form/index.js [hunk 17]
```diff
@@ -0,0 +1,158 @@
...
+        })
+      });
+      const content = await res.json();
+      if (!content.error) {
+        window.location.assign('/super-admins');
+      } else {
+        setFormModal(true);
+        setErrorMessage(content.message);
+      }
+      console.log(content);
```
Comment (lmalvasia) [inline src/Components/SuperAdmins/Form/index.js:96]: Remove this console log

File: src/Components/SuperAdmins/Form/index.js [hunk 18]
```diff
@@ -0,0 +1,158 @@
...
+    }
+  };
+
+  return (
+    <section className={styles.container}>
+      <div className={styles.flex}>
+        <Title modalTitle={modalTitle} className={styles.title} />
```
Comment (lmalvasia) [inline src/Components/SuperAdmins/Form/index.js:103]:
```suggestion
        <Title modalTitle={isEditing ? 'Edit super admin' : 'Create super admin'} className={styles.title} />
```

File: src/Components/SuperAdmins/Form/ModalTitle/index.js [hunk 19]
```diff
@@ -0,0 +1,5 @@
+function Title(props) {
+  return <h4>{props.modalTitle}</h4>;
```
Comment (lmalvasia) [inline src/Components/SuperAdmins/Form/ModalTitle/index.js:2]: It's necessary this component? I think that we could move this to the form component directly 😂  (It's just a h4)

File: src/Components/SuperAdmins/List/index.js (hunk 1 above)
Comment (camilavaleriarojas) [reply src/Components/SuperAdmins/List/index.js:51]: Got it!

**Review by lmalvasia — APPROVED** [review]

**Pull request merged**